checks/normandy/jexl_error_rate.py

```diff
@@ -21,13 +24,29 @@
     min_timestamp = min(r["min_timestamp"] for r in rows)
     max_timestamp = max(r["max_timestamp"] for r in rows)
 
-    total = sum(row["total"] for row in rows)
-    classify_errors = sum(
-        row["total"] for row in rows if row["status"] == "content_error"
-    )
-    error_rate = classify_errors * 100.0 / total
+    # The Redash query returns statuses by periods (eg. 10min).
+    # First, agregate totals by period.
+    periods: Dict[Tuple[str, str], Dict] = {}
+    for row in rows:
+        period: Tuple[str, str] = (row["min_timestamp"], row["max_timestamp"])
+        if period not in periods:
+            periods[period] = defaultdict(int)
+        status = row["status"]
+        periods[period][status] += row["total"]
+
+    # Then, keep the period with highest error rate.
+    max_error_rate = 0.0
+    for period, all_statuses in periods.items():
+        total = sum(all_statuses.values())
+        classify_errors = all_statuses.get("content_error", 0)
+        error_rate = classify_errors * 100.0 / total
+        if max_error_rate < error_rate:
+            max_error_rate = error_rate
```

Comment:
Instead of the if block, what about this?

```python
max_error_rate = max(max_error_rate, error_rate)
```

Reply:
Aww.. of course :roll_eyes: